### ISSUE TITLE
Bug: 'structuredClone' is not defined with env 'browser'

### ISSUE THREAD

**Issue by regseb** [eslint/eslint#15859]
### Environment

Node version: v18.1.0
npm version: 8.8.0
Local ESLint version: 8.15.0
Global ESLint version: none
Operating System: Ubuntu 20.04.4 LTS

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```JSON{
    "env": {
        "browser": true
    },
    "rules": {
        "no-undef": "error"
    }
}
```
</details>

```js
console.log(structuredClone({ foo: "bar" }));
```

### What did you expect to happen?

No errors reported.

### What actually happened?

```
/home/regseb/testcase/index.js
  1:13  error  'structuredClone' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

[`structuredClone`](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) is a function specified in the HTML Standard.

**Comment by mdjermanovic** [comment]
Hi @regseb, thanks for the issue!

For environments, we are using the [globals](https://www.npmjs.com/package/globals) package. If some variables are missing, you can open an issue/PR here:

https://github.com/sindresorhus/globals

When the variables are added and a new version is released, we could update the dependency.

**Comment by regseb** [comment]
Thanks for the feedback. I created the pull request https://github.com/sindresorhus/globals/pull/190
